### PR TITLE
feat(AWS Lambda): Change default hashing algorithm

### PR DESCRIPTION
**Pull request by pgrzesik**
BREAKING CHANGE: Default lambda hashing algorithm was changed to `20201221`

### CODE REVIEW

**Review by mnapoli — APPROVED** [review]
👍 , might be good to get that in early next week if users deploy anything with the v3 beta

File: docs/guides/upgrading-v3.md [hunk 1]
```diff
@@ -210,7 +210,7 @@ However, we highly encourage an upgrade to the new algorithm. To upgrade, you mu
 **NOTE**: Please keep in mind that these changes require two deployments with manual configuration adjustment between them. It also creates two additional versions and temporarily overrides descriptions of your functions. Migration will need to be done separately for each of your environments/stages.
 
 1. Run `sls deploy` with additional `--enforce-hash-update` flag: that flag will override the description for Lambda functions, which will force the creation of new versions.
-2. Set `provider.lambdaHashingVersion` to `20201221` in your configuration: your service will now always deploy with the new Lambda version hashes (which is the new defualt in v3).
+2. Set `provider.lambdaHashingVersion` to `20201221` in your configuration: your service will now always deploy with the new Lambda version hashes (which is the new default in v3).
```
Comment (medikoo) [inline docs/guides/upgrading-v3.md:213]: I think this change should land in `master`, so online docs are updated

File: lib/plugins/aws/package/compile/functions.js [hunk 2]
```diff
@@ -478,10 +477,8 @@ class AwsCompileFunctions {
       delete functionProperties.Tags;
 
       const lambdaHashingVersion = this.serverless.service.provider.lambdaHashingVersion;
-      if (lambdaHashingVersion >= 20201221 || enforceHashUpdate) {
-        functionProperties.layerConfigurations = layerConfigurations;
-        versionHash.write(JSON.stringify(deepSortObjectByKey(functionProperties)));
-      } else {
+      // TODO: TESTS
```
Comment (medikoo) [inline lib/plugins/aws/package/compile/functions.js:480]: Same here (?)

File: docs/guides/upgrading-v3.md (hunk 1 above)
Comment (pgrzesik) [reply docs/guides/upgrading-v3.md:213]: yes, this change should be in `master`, I'll remove it from this PR

File: lib/plugins/aws/package/compile/functions.js (hunk 2 above)
Comment (pgrzesik) [reply lib/plugins/aws/package/compile/functions.js]: Removed, this is a leftover comment

**Review by medikoo — APPROVED** [review]
Looks great 👍

**Pull request merged**